README.md

```diff
@@ -4,49 +4,100 @@
+
+### Custom configuration
+Custom configuration is only required when using a different set of span decoratos.
 
-Custom configuration 
 ```java
-// code sample from javax.ws.rs.core.Application#getSingletons();
-DynamicFeature tracing = new ServerTracingDynamicFeature.Builder(tracer)
-    .withDecorators(decorators)
-    .build();
-singletons.add(tracing);
-return singletons;
+// code sample from javax.ws.rs.core.Application
+public Set<Object> getSingletons() {
+  DynamicFeature tracing = new ServerTracingDynamicFeature.Builder(tracer)
+      .withDecorators(decorators)
+      .build();
+
+  return Collections.singleton(tracing);
+}
+
 ```
             
 An example of traced REST endpoint:
 ```java
 @GET
 @Path("/hello")
 @Traced(operationName = "helloRenamed") // optional, see javadoc
-public Response hello(@BeanParam ServerSpanContext serverSpanContext) { // optional to get server span context
-    /**
-     * Some business logic
-     */
-    Span childSpan = tracer.buildSpan("businessOperation")
-            .asChildOf(serverSpanContext.get())
-            .start())
-    childSpan.finish();
-
-    return Response.status(Response.Status.OK).build();
+public Response hello(@BeanParam TracingContext tracingContext) { // optional to get server span context
+
+  // this span will be ChildOf of span representing server request processing, if it is not async, keep reading!
+  Span childSpan = tracer.buildSpan("businessOperation")
+          .start())
+
+   // business logic
+  childSpan.finish();
+
+  return Response.status(Response.Status.OK).build();
 }
 ```
 
-## Tracing Client Requests
+#### Async handlers
+In case of async handlers server span is not accessible via `tracer.activeSpan()`, however span context
+can be obtained via `@BeanParam TracingContext tracingContext`. This object is injected as a handler param.
+
+## Tracing client requests
 ```java
-Client client = ClientBuilder.newClient();
-client.register(ClientTracingFeature.class);
-
-Response response = jaxRsClient.target("http://localhost/endpoint")
-    .request()
-    .property(TracingProperties.CHILD_OF, parentSpanContext) // optional, by default new trace is started
-    .property(TracingProperties.TRACING_DISABLED, false) // optional, by default false
-    .get();
+Client client = ClientBuilder.newBuilder()
+  .reqister(ClientTracingFeature.class)
+  .build();
+
+Response response = client.target("http://localhost/endpoint")
+  .request()
+  .property(TracingProperties.CHILD_OF, parentSpanContext) // optional, by default new parent is inferred from span source
+  .property(TracingProperties.TRACING_DISABLED, false) // optional, by default everything is traced
+  .get();
+```
+
+### Async
+Async requests are executed in a different thread than when the client has been invoked, therefore
+spans representing client requests are not connected to appropriate parent. To fix this JAX-RS client
+has to use OpenTracing-aware `ExecutorService`.
```

Comment:
Link added, actually that dependency is pulled by default so users would find it.